### PR TITLE
Feature new student orgs

### PR DESCRIPTION
**Pull request by drewvolz**
This creates a grouping for new student organizations to be featured at the top of the orgs list.

orgs

### CODE REVIEW

File: source/views/student-orgs/list.js [hunk 1]
```diff
@@ -77,9 +77,20 @@ export class StudentOrgsView extends React.Component {
           $groupableName: head(startCase(sortableName)),
         }
       })
+      // All orgs
       let sorted = sortBy(withSortableNames, '$sortableName')
       let grouped = groupBy(sorted, '$groupableName')
-      this.setState({orgs: grouped})
+      // New orgs
+      let newOrgs = sorted.filter(org => org.newOrg)
+      let newOrgsSortableNames = map(newOrgs, item => {
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: Join this into one statement, if you would?

sorted
  .filter()
  .map()

**Review by hawkrives — CHANGES_REQUESTED** [review]
One requested change, and one comment: would you look into AlphabetListView and see if you can specify the letters on the right side separately?  IIRC, you can specify an "extractor" function which will produce the items used for the side; you could do `$groupableName === "New" ? "•" : $groupableName[0]` or something to change "New" into a single char.

**Comment by drewvolz** [comment]
I will look into how to specify groupnames explicitly.

**Comment by hawkrives** [comment]
We could probably call it "• New" and leave it how it's currently taking the first letter of each item, if we wanted.

**Edit:** Nevermind. That doesn't work, because the section headings and the sidebar use the same labels in the current implementation.

**Comment by hawkrives** [comment]
Given https://github.com/sunnylqm/react-native-alphabetlistview#getsectiontitle, I think that the `getSectionTitle` prop is the the ListSectionHeader, and `getSectionListTitle` is for the things on the right.

**Comment by drewvolz** [comment]
Done! Thanks for your help.

**Review by hawkrives — CHANGES_REQUESTED** [review]
You have cellHeight in here twice, accidentally.

File: source/views/student-orgs/list.js [hunk 2]
```diff
@@ -184,6 +199,7 @@ export class StudentOrgsView extends React.Component {
         data={this.state.orgs}
         cell={this.renderRow}
         // just setting cellHeight sends the wrong values on iOS.
+        getSectionListTitle={this.getSectionListTitle}
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: Please move this line above the comment

File: source/views/student-orgs/list.js [hunk 3]
```diff
@@ -119,6 +129,11 @@ export class StudentOrgsView extends React.Component {
     )
   };
 
+  getSectionListTitle = (...args: any) => {
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: `= (name: string) =>` and get rid of the array access

File: source/views/student-orgs/list.js [hunk 4]
```diff
@@ -83,9 +83,19 @@ export class StudentOrgsView extends React.Component {
...
-      this.setState({orgs: grouped})
+      // New orgs
+      let newOrgs = sorted.filter(org => org.newOrg).map(item => {
+        return {
+          ...item,
+          $groupableName: 'New',
+        }
+      })
+      let newOrgsGrouped = groupBy(newOrgs, '$groupableName')
+      let combined = {...newOrgsGrouped, ...grouped}
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: Please inline this into the setState call (setState({orgs: {...newOrgs, ...allOrgs}})

File: source/views/student-orgs/list.js [hunk 5]
```diff
@@ -83,9 +83,19 @@ export class StudentOrgsView extends React.Component {
           $groupableName: head(startCase(sortableName)),
         }
       })
+      // All orgs
       let sorted = sortBy(withSortableNames, '$sortableName')
       let grouped = groupBy(sorted, '$groupableName')
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: Please rename this to allOrgsGrouped

File: source/views/student-orgs/list.js [hunk 6]
```diff
@@ -83,9 +83,19 @@ export class StudentOrgsView extends React.Component {
...
-      this.setState({orgs: grouped})
+      // New orgs
+      let newOrgs = sorted.filter(org => org.newOrg).map(item => {
+        return {
+          ...item,
+          $groupableName: 'New',
+        }
```
Comment (hawkrives) [inline source/views/student-orgs/list.js]: Could you just put this whole thing on one line please? return through the end bracket.

**Review by hawkrives — CHANGES_REQUESTED** [review]

**Comment by hawkrives** [comment]
Can you please merge master into this branch and run prettier on it?

Edit: I ran prettier.

**Comment by hawkrives** [comment]
So the current implementation goes like this:

```js
// All orgs
let sorted = sortBy(withSortableNames, '$sortableName')
let allOrgsGrouped = groupBy(sorted, '$groupableName')
// New orgs
let newOrgs = sorted.filter(org => org.newOrg).map(item => {
  return {...item, $groupableName: 'New'}
})
let newOrgsGrouped = groupBy(newOrgs, '$groupableName')
let combined = {...newOrgsGrouped, ...allOrgsGrouped}
this.setState({orgs: combined})
```

What about changing it to this?

```js
let sorted = sortBy(withSortableNames, '$sortableName')
let allOrgsGrouped = groupBy(sorted, '$groupableName')

let newOrgs = sorted.filter(org => org.newOrg)
let orgs = {New: newOrgs, ...allOrgsGrouped}

this.setState({orgs: orgs})
```

**Review by hawkrives — APPROVED** [review]

**Comment by drewvolz** [comment]
That particular redesign resulted in every org being marked as new. Also, the font size for that new icon has decreased.

**Comment by hawkrives** [comment]
I think you commented on the wrong PR?

**Comment by drewvolz** [comment]
I meant to comment here as I thought this PR had something to do with it... although it was introduced in [#830 Add new warnings to linter](https://github.com/StoDevX/AAO-React-Native/pull/830)

**Pull request merged**